interfaces/builtin/uhid.go

```diff
@@ -0,0 +1,40 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const uhidConnectedPlugAppArmor = `
+# Description: Allows accessing the UHID to create kernel
+#  hid devices from user-space.
+
+  # Requires CONFIG_UHID
+  /dev/uhid                       rw,
+`
+
+func NewUhidInterface() interfaces.Interface {
+	return &commonInterface{
+		name: "uhid",
+		connectedPlugAppArmor: uhidConnectedPlugAppArmor,
+		reservedForOS:         true,
+	}
+}
```

Comment:
You are also going to want to add the udev backend considering my comment here: https://github.com/snapcore/snapd/pull/2459/files#r92240828.

We haven't been consistent on this across all interfaces which is something that should be addressed in future commits, but for uhid it seems very likely that it will be used in combination with something like serial-port, etc so using the udev backend is key for making it all work together.